src/Simplex/Chat.hs

```diff
@@ -1577,22 +1587,29 @@ setAllExpireCIFlags b = do
     keys <- M.keys <$> readTVar expireFlags
     forM_ keys $ \k -> TM.insert k b expireFlags
 
-deleteFile :: forall m. ChatMonad m => User -> CIFileInfo -> m ()
+deleteFile :: forall m. ChatMonad m => User -> CIFileInfo -> m [AgentConnId]
 deleteFile user fileInfo = deleteFile' user fileInfo False
 
-deleteFile' :: forall m. ChatMonad m => User -> CIFileInfo -> Bool -> m ()
-deleteFile' user CIFileInfo {filePath, fileId, fileStatus} sendCancel =
-  (cancel' >> delete) `catchError` (toView . CRChatError (Just user))
+deleteFile' :: forall m. ChatMonad m => User -> CIFileInfo -> Bool -> m [AgentConnId]
+deleteFile' user CIFileInfo {filePath, fileId, fileStatus} sendCancel = do
+  aConnIds <- case fileStatus of
+    Just fStatus -> cancel' fStatus `catchError` (\e -> toView (CRChatError (Just user) e) >> pure [])
```

Comment:
can't cancel throw and have some connections to delete? if it can't throw then why have catch here as well?